mavros_extras/src/plugins/obstacle_avoidance.cpp

```diff
@@ -68,167 +67,263 @@ class ObstacleAvoidancePlugin : public plugin::PluginBase {
-        obstacle_avoidance.point_4[1] = req->point_4.position.x;
-        obstacle_avoidance.point_4[2] = -req->point_4.position.z;
-        obstacle_avoidance.point_4[3] = req->point_4.velocity.y;
-        obstacle_avoidance.point_4[4] = req->point_4.velocity.x;
-        obstacle_avoidance.point_4[5] = -req->point_4.velocity.z;
-        obstacle_avoidance.point_4[6] = req->point_4.acceleration_or_force.y;
-        obstacle_avoidance.point_4[7] = req->point_4.acceleration_or_force.x;
-        obstacle_avoidance.point_4[8] = -req->point_4.acceleration_or_force.z;
-        obstacle_avoidance.point_4[9] = wrap_pi(-req->point_4.yaw + (M_PI / 2.0f));
-        obstacle_avoidance.point_4[10] = req->point_4.yaw_rate;
-
-        obstacle_avoidance.point_5[0] = req->point_5.position.y;
-        obstacle_avoidance.point_5[1] = req->point_5.position.x;
-        obstacle_avoidance.point_5[2] = -req->point_5.position.z;
-        obstacle_avoidance.point_5[3] = req->point_5.velocity.y;
-        obstacle_avoidance.point_5[4] = req->point_5.velocity.x;
-        obstacle_avoidance.point_5[5] = -req->point_5.velocity.z;
-        obstacle_avoidance.point_5[6] = req->point_5.acceleration_or_force.y;
-        obstacle_avoidance.point_5[7] = req->point_5.acceleration_or_force.x;
-        obstacle_avoidance.point_5[8] = -req->point_5.acceleration_or_force.z;
-        obstacle_avoidance.point_5[9] = wrap_pi(-req->point_5.yaw + (M_PI / 2.0f));
-        obstacle_avoidance.point_5[10] = req->point_5.yaw_rate;
-
-        std::copy(req->point_valid.begin(), req->point_valid.end(), obstacle_avoidance.point_valid.begin());
-        std::copy(req->field_of_view.begin(), req->field_of_view.end(), obstacle_avoidance.field_of_view.begin());
-
-        UAS_FCU(m_uas)->send_message_ignore_drop(obstacle_avoidance);
+		mavlink::common::msg::OBSTACLE_AVOIDANCE obstacle_avoidance {};
+		obstacle_avoidance.time_usec = req->header.stamp.toNSec() / 1000;	//!< [milisecs]
+		obstacle_avoidance.type = req->type;	//!< trajectory type (waypoints, bezier)
+
+		/* Transformation from ENU to NED */
+		// [[[cog:
+		// for p in ('1', '2', '3', '4', '5'):
+		//     cog.outl("Eigen::Vector3d position_point_%s, velocity_point_%s, acceleration_point_%s;" % (p, p, p))
+		//     cog.outl("tf::pointMsgToEigen(req->point_%s.position, position_point_%s);" % (p, p))
+		//     cog.outl("tf::vectorMsgToEigen(req->point_%s.velocity, velocity_point_%s);" % (p, p))
+		//     cog.outl("tf::vectorMsgToEigen(req->point_%s.acceleration_or_force, acceleration_point_%s);" % (p, p))
+		//     cog.outl("position_point_%s = ftf::transform_frame_enu_ned(position_point_%s);" % (p, p))
+		//     cog.outl("velocity_point_%s = ftf::transform_frame_enu_ned(velocity_point_%s);" % (p, p))
+		//     cog.outl("acceleration_point_%s = ftf::transform_frame_enu_ned(acceleration_point_%s);" % (p, p))
+		//     cog.outl("double yaw_ned_point_%s = wrap_pi(-req->point_%s.yaw + (M_PI / 2.0f));;" % (p, p))
```

Comment:
Use `ftf::to_eigen()` 	https://github.com/mavlink/mavros/blob/master/mavros/include/mavros/frame_tf.h#L414 instead of `tf::`.
That allow chain functions.

Add white string between iterations.